### PR TITLE
Doc schema.registry support for JSON 

### LINKED ISSUE #1218: Document: feat(source): support json schema
This issue tracks the documentation update needed for the merged PR #11797.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/11797
Source PR Merged At: 2023-08-29T05:26:52Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Doc schema.registry support for plain JSON format

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/11797

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1218

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by wugouzi — APPROVED** [review]
LGTM. The pr also supports maxwell, debezium and canal in theory but there are no related tests.

File: docs/sql/commands/sql-create-source.md [hunk 1]
```diff
@@ -188,13 +188,13 @@ ENCODE AVRO (
...
-RisingWave decodes JSON directly from external sources. When creating a source from streams in JSON, you need to define the schema of the source within the parentheses after the source name, and specify the data and encoding formats in the `FORMAT` and `ENCODE` sections. You can directly reference data fields in the JSON payload by their names as column names in the schema.
+RisingWave decodes JSON directly from external sources. When creating a source from streams in JSON, you can define the schema of the source within the parentheses after the source name or specify a `schema.registry`. Specify the data and encoding formats in the `FORMAT` and `ENCODE` sections. You can directly reference data fields in the JSON payload by their names as column names in the schema.
 
 Syntax:
 
 ```sql
-FORMAT PLAIN
-ENCODE JSON
+FORMAT PLAIN 
+ENCODE JSON [ (schema.registry = 'schema_registry_url') ]
```
Comment (tabVersion) [inline docs/sql/commands/sql-create-source.md]: we also have `schema.registry.username` and `schema.registry.password` here for auth use, please confirm @wugouzi

File: docs/sql/commands/sql-create-source.md [hunk 2]
```diff
@@ -285,13 +285,15 @@ ENCODE JSON
...
+When consuming data in JSON from Kafka topics, the `FORMAT` and `ENCODE` sections need to be specified as `UPSERT` and `JSON` respectively. RisingWave will be aware that the source message contains key fields as primary columns, as well as the Kafka message value field. If the value field of the message is not null, the row will be updated if the message key is not empty and already exists in the database table, or inserted if the message key is not empty but does not exist yet in the database table. If the value field is null, the row will be deleted. 
+
+You can define the schema of the source within the parentheses after the source name or specify a `schema.registry`.
 
 Syntax:
 
 ```sql
 FORMAT UPSERT
-ENCODE JSON
+ENCODE JSON [ (schema.registry = 'schema_registry_url') ]
```
Comment (tabVersion) [inline docs/sql/commands/sql-create-source.md]: ditto

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/sql/commands/sql-create-source.md (hunk 1 above)
Comment (wugouzi) [reply docs/sql/commands/sql-create-source.md]: Yes, thanks. I'll modify the desc of that pr too.

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**